src/js/stores/middlewares/messages.js

```diff
@@ -0,0 +1,24 @@
+import { ADD_MESSAGE } from 'actions/conversation-actions';
+import { showSettingsNotification } from 'actions/app-state-actions';
+
+
+export function firstMessage(store) {
+    return function(next) {
+        return function(action) {
+            let result = next(action);
+            let state = store.getState();
+            if (state.appState.settingsEnabled && action.type === ADD_MESSAGE && action.message.role === 'appUser' && !state.user.email) {
```

Comment:
Seems odd that we have to handle first app user messages with a middleware.

Is there a way we could dispatch a `FIRST_USER_MESSAGE` event? Maybe do it from `conversation-service`? Then it's up to the settings view to listen for this event and decide whether or not to fire.